lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1027,14 +1033,26 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z"):
             dict(x=0, y=1, z=2), vertical_axis=vertical_axis
         )
 
-    def set_proj_type(self, proj_type):
+    def set_proj_type(self, proj_type, focal_length=None):
         """
         Set the projection type.
 
         Parameters
         ----------
         proj_type : {'persp', 'ortho'}
-        """
+            The projection type.
+        focal_length : float, default: None
+            For a projection type of 'persp', the focal length of the virtual
+            camera. If None, defaults to 1.
+            The focal length can be computed from a desired Field Of View via
+            the equation: focal_length = 1/tan(FOV/2)
+        """
+        self.focal_length = None
+        if proj_type == 'persp':
+            self.focal_length = focal_length
+            if focal_length is None:
+                self.focal_length = 1
```

Comment:
Perhaps it should also error if the projection type is not 'persp' and focal_length isn't `None` (unless a focal length makes sense in 'ortho'?)

Reply:
Physically, you can have a lens with negative focal length (a diverging lens), though I don't think it really makes sense for a camera and expect it'll look really weird. Maybe it'll just flip the image? I'm away from my home setup for the week and have been doing this on github's in-browser vscode (nifty stuff!), so I'll have to wait to really play with it.

Reply:
It doesn't have a meaning (and isn't used) for an orthographic projection, we can error it if we want but I don't think it's too much of a foot-gun since it'll just get ignored.

Reply:
Posted an image of this